### DISCUSSION TITLE
Query related to question set editor installation

### DISCUSSION THREAD

**Discussion by rajputakku** [Sunbird-inQuiry/Community#71] [Q&A]
Hi team,

We're trying to setup [Sunbird inQuiry](https://inquiry.sunbird.org/learn/readme) by following the instructions [here](https://inquiry.sunbird.org/use/developer-installation). There are 3 components that are mentioned in the doc:

1. [Question & Question Set Service](https://inquiry.sunbird.org/use/developer-installation/question-and-question-set-service)
2. [Question Set Editor](https://inquiry.sunbird.org/use/developer-installation/question-set-editor)
3. [Question Set Player](https://inquiry.sunbird.org/use/developer-installation/question-set-player)

We've successfully installed the Question & Question Set Service and are also able to create QuestionSet & Questions using the APIs.

After [installing](https://inquiry.sunbird.org/use/developer-installation/question-set-editor/installation) the Question Set Editor, we're not able to see any questions in the UI. Our expectation was to see a form like the one shown [here](https://inquiry.sunbird.org/use/developer-installation/question-set-editor/installation/sample-form-configuration#root-node-form-2). Instead, we're getting the following blank screen:

[main page](https://drive.google.com/file/d/1XODCz4NBTQ9xqORWPhDy2fwtpqU5SIKS/view?usp=sharing)
[question editor page](https://drive.google.com/file/d/1gNiqdR-DN3oTJyxddI_8h8-laprstpWJ/view?usp=sharing)

We're also seeing some failed API requests in the logs:
[editor console1](https://drive.google.com/file/d/1DW5Q8-AgtWtxWk2U3kqtYqBRh-OLsyDZ/view?usp=sharing)
[editor console 2](https://drive.google.com/file/d/1PXir88TPc7oXm7pHu0Idv6xSTd_HmEsZ/view?usp=sharing)
[assessment service console](https://drive.google.com/file/d/12pfrJiQR8TrfX_VoMDpwfpNc8uhK1sCG/view?usp=sharing)

We want to know how can we setup Sunbird inQuiry. Is there any other service that we need to install to make this work end to end ?

**Comment by sajeshkayyath** [comment]
Thanks for reaching out @rajputakku 
We will have a check and get back to you

@krgauraw / @vaibhavbhuva / @rajnishdargan / @PoojaJManjooran / @kameshbhr / @surendrasinghs / @pallakartheekreddy

**Comment by sajeshkayyath** [comment]
@rajputakku - a few followup questions on this issue that you were facing.

I see from the [assessment service console](https://drive.google.com/file/d/12pfrJiQR8TrfX_VoMDpwfpNc8uhK1sCG/view?usp=sharing) there is 2 reported error stating "Entry is not found in Cassandra for content with the identifier : do_11374869021013606411" and later part of the logs indicate that the fetch was successful. 

Can you also help us with what errors you see in the browser console and network tab? Please provide the details in a text file for us to go through the logs closely.

Awaiting your reply.

cc @rajnishdargan / @vaibhavbhuva / @krgauraw / @PoojaJManjooran / @kameshbhr / @pallakartheekreddy

**Comment by rajputakku** [comment]
Hi @sajeshkayyath Here i am attaching few more files for your further references:
[console error text file](https://drive.google.com/file/d/13Ul-CqLr1zNHU6KKvsalVmrynolcR11U/view?usp=share_link)
[console error screenshot](https://drive.google.com/file/d/160Tgr97OVXcNYyU7rdAKS3VxOqfnY8J5/view?usp=share_link)
[network error screenshot](https://drive.google.com/file/d/1d74lSuc0lftiaZMUSWKpoqbLDP_4iLAV/view?usp=share_link)
[editor console 1](https://drive.google.com/file/d/1I6BfFWfSXFCwmXGVwAIUb3mlpb65cBrE/view?usp=share_link)
[editor console 2](https://drive.google.com/file/d/1GKYepiSFzJ2L7KOFJ7lVkE1ddsC9k8j0/view?usp=share_link)

**Reply by sajeshkayyath** [reply]
@rajputakku - from the above logs, we see that you may not be having the following dependent services running locally.

- content-service
- taxonomy-service
- search-service
- learning-service
- data-pipeline

This is all part of knowlg: https://knowlg.sunbird.org/

We are also getting the single page documentation that can help you in setting up all the dependent service locally and once we have it, we will share it with you.

In the meantime, for getting the editor working in a standalone fashion, there is a possibility to mock the end points for dependent services. You can try this and in the meantime, we are trying to formulate a step in achieving the same and will share it with you.  

If you only wanted to try out the editor capability this link has a demo application: https://dev.inquiry.sunbird.org/ which will give you an overview.

cc @krgauraw / @rajnishdargan / @vaibhavbhuva / @pallakartheekreddy / @PoojaJManjooran / @kameshbhr / @surendrasinghs

**Reply by sajeshkayyath** [reply]
@rajputakku - a followup to see if you were able to try the installation instruction from Knowlg. 

Requesting you to follow the documentation given [here](https://knowlg.sunbird.org/learn/readme) and use the [discussion forum](https://github.com/orgs/Sunbird-Knowlg/discussions) for following up on the queries related to Knowlg.

@pallakartheekreddy / @PoojaJManjooran / @krgauraw / @rajnishdargan / @vaibhavbhuva / @kameshbhr

**Reply by rajputakku** [reply]
Hi @sajeshkayyath  thank you for providing all the resources, i will get back to you if i get stuck anywhere.

**Reply by rajputakku** [reply]
Hi @sajeshkayyath 
I have a few queries that I would like to discuss with you regarding your editor platform's form creation and integration capabilities.
Currently, we have integrated [Jotform](https://www.jotform.com/) into our platform, which enables us to create forms in multiple languages (such as Hindi and Marathi) and add various question types, including single and multiple choice, image checkboxes, and audio files in the assessment forms. As we have already created several live forms on our platform, we are looking for a compatible solution to migrate our existing forms.

Thus, I would like to inquire whether your platform can provide similar form creation and integration facilities. To give you a better understanding of our requirements, I have attached a few reference forms from Jotforms [jotform example 1](https://form.jotform.com/220752237822453) [jotform example 2](https://form.jotform.com/223518731019453)

Additionally, I have a query about the mandatory fields (such as Board/Syllabus, Medium etc) in your question set. Are they fixed or dynamic? As we do not require all the information, it would be helpful to know whether we can customise the mandatory fields to only keep the required ones. If your platform allows for self-creation of forms with customizable mandatory fields, that would be ideal.

I look forward to hearing from you.

cc @krgauraw / @rajnishdargan / @vaibhavbhuva / @pallakartheekreddy / @PoojaJManjooran / @kameshbhr / @surendrasinghs

**Reply by sajeshkayyath** [reply]
@rajputakku - Can you post this additional requirement as a new thread for traceability? Please mark this thread as closed if the original query is addressed. 

Looking forward for a new discussion item and in the meantime, we are having a check from our end.

**Reply by rajputakku** [reply]
Hey @sajeshkayyath Sure, creating a new thread for this requirement. Please find below the new thread link:
https://github.com/orgs/Sunbird-inQuiry/discussions/75

**Reply by rajputakku** [reply]
Hey @sajeshkayyath I merely wanted to keep this ticket open so that I could subsequently ask questions about the installation of the editor.
